### DISCUSSION TITLE
Syntax error specifying `fail-fast: false`

### DISCUSSION THREAD

**Discussion by lucywyman** [community/community#26602] [Actions]
<p>Hello! I was recently adding a matrix to our Github Actions, and when I tried to set <code>fail-fast</code> to false like so:</p>
<pre><code class="lang-auto">  test:
    name: Linux tests
    runs-on: ubuntu-latest
    strategy:
      fail-fast: false
      matrix:
        ruby: [2.7, 3.0]
    steps:
...
</code></pre>
<p>It raise the following error in Github Actions:<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/1/2/12976bcd515dc0a9a16b71b97df6695c02c48886.png" data-download-href="/uploads/short-url/2Et0MKw0WcXEFY2eDQcsPeo1kbk.png?dl=1" title="fail-fast-false" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/5055819/181098641-a223fa03-d28f-40cb-9152-80e89499dd59.png" alt="fail-fast-false" data-base62-sha1="2Et0MKw0WcXEFY2eDQcsPeo1kbk" width="690" height="452" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/1/2/12976bcd515dc0a9a16b71b97df6695c02c48886_2_690x452.png, https://aws1.discourse-cdn.com/github/optimized/3X/1/2/12976bcd515dc0a9a16b71b97df6695c02c48886_2_1035x678.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/1/2/12976bcd515dc0a9a16b71b97df6695c02c48886_2_1380x904.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/1/2/12976bcd515dc0a9a16b71b97df6695c02c48886_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">fail-fast-false</span><span class="informations">1633×1071 66.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
It seems like this is directly related to fail-fast, but I’m not sure why it won’t accept false when <code>true</code> is the default?

**Comment by Simran-B** [comment, marked as answer]
<p><code>fail-fast</code> was incorrectly placed under <code>matrix</code> here:</p><aside class="onebox githubfolder">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/puppetlabs/bolt/actions/runs/733806628/workflow#L25" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:400/400;"><img src="https://avatars.githubusercontent.com/u/234268?s=400&amp;amp;v=4" class="thumbnail" width="400" height="400"></div>

<h3><a href="https://github.com/puppetlabs/bolt/actions/runs/733806628/workflow#L25" target="_blank" rel="noopener nofollow ugc">puppetlabs/bolt</a></h3>

<p><a href="https://github.com/puppetlabs/bolt/actions/runs/733806628/workflow#L25" target="_blank" rel="noopener nofollow ugc">ns/runs/733806628/workflow</a></p>

  <p><span class="label1">Bolt is an open source orchestration tool that automates the manual work it takes to maintain your infrastructure on an as-needed basis or as part of a greater orchestration workflow. It can be ins...</span></p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>The parent key needs to be <code>strategy</code>.</p>
<p>If you put it under <code>matrix</code> instead, then it is treated like any other user-defined matrix option with no special meaning. GitHub expects an array however, which explains the error message. If you passed <code>[false]</code> then it would be accepted as matrix option, but the combinations “Ruby 2.7, false” and “Ruby 3.0, false” aren’t meaningful.</p>

**Comment by lucywyman** [comment]
<p>:woman_facepalming: Oh my goodness, thank you!</p>
